src/main/java/sirius/kernel/cache/CacheRemoverBuilder.java

```diff
@@ -0,0 +1,123 @@
+ * @see Cache#addRemover(String, BiPredicate)
+ */
+public interface CacheRemoverBuilder<K, V, T> {
+    /**
+     * Similar to {@link java.util.stream.Stream#map Stream.map}, this method applies a mapping function to the
+     * objects this {@link CacheRemoverBuilder} operates on.
+     *
+     * @param mapper The mapping function that is applied
+     * @param <R>    The new underlying type that replaces T
+     * @return a new {@link CacheRemoverBuilder} with the mapped underlying type
+     */
+    <R> CacheRemoverBuilder<K, V, R> map(BiFunction<String, T, R> mapper);
+
+    /**
+     * Convenience method for {@link #map(BiFunction)} ignoring the test parameter.
+     *
+     * @param mapper The mapping function that is applied
+     * @param <R>    The new underlying type that replaces T
+     * @return a new {@link CacheRemoverBuilder} with the mapped underlying type
+     */
+    default <R> CacheRemoverBuilder<K, V, R> map(Function<T, R> mapper) {
+        return map((string, value) -> mapper.apply(value));
+    }
+
+    /**
+     * Similar to {@link java.util.stream.Stream#filter Stream.filter}, this method filters the objects this
+     * {@link CacheRemoverBuilder} operates on. All objects that do not match the predicate are ignored, and
+     * will not be removed from the cache. This allows to discard those objects early.
+     *
+     * @param predicate The predicate to test the objects
+     * @return A new {@link CacheRemoverBuilder} operating on the remaining objects
+     */
+    CacheRemoverBuilder<K, V, T> filter(BiPredicate<String, T> predicate);
+
+    /**
+     * Convenience method for {@link #filter(BiPredicate)} ignoring the test parameter.
+     *
+     * @param predicate The predicate to test the objects
+     * @return A new {@link CacheRemoverBuilder} operating on the remaining objects
+     */
+    default CacheRemoverBuilder<K, V, T> filter(Predicate<T> predicate) {
+        return filter((string, value) -> predicate.test(value));
+    }
+
+    /**
+     * This method serves as a counterpart for {@link #filter}. The objects that match the predicate will be
+     * removed from the cache without further processing.
+     *
+     * @param predicate The predicate to test the objects
+     * @return A new {@link CacheRemoverBuilder} operating on the remaining objects
+     */
+    CacheRemoverBuilder<K, V, T> removeAlways(BiPredicate<String, T> predicate);
+
+    /**
+     * Convenience method for {@link #removeAlways(BiPredicate)} ignoring the test parameter.
+     *
+     * @param predicate The predicate to test the objects
+     * @return A new {@link CacheRemoverBuilder} operating on the remaining objects
+     */
+    default CacheRemoverBuilder<K, V, T> removeAlways(Predicate<T> predicate) {
+        return removeAlways((string, value) -> predicate.test(value));
+    }
+
+    /**
+     * This method terminates the builder. All objects that are not yet ignored via {@link #filter} or
```

Comment:
excellent javadoc but the summary of the method would be "This method terminates the builder."

We commonly use a singe sentence in a single line for a short description - followed (if necessarry) by a longer description.